sdks/python/apache_beam/runners/worker/data_plane.py

```diff
@@ -292,8 +357,15 @@ def close_callback(data):
               instruction_id=instruction_id,
               transform_id=transform_id,
               data=b''))
-    return ClosableOutputStream(
-        close_callback, flush_callback=add_to_send_queue)
+
+    if self._data_buffer_time_limit > 0:
+      return TimeBasedBufferingClosableOutputStream(
```

Comment:
Continuing on the comment above, maybe make a helper to avoid logic duplication.